### PR TITLE
Reorganize in a table the different stats output by the "statistics by categories" alg

### PR DESCRIPTION
**Pull request by DelazJ**
based on field type
Also adds some tweaks to basic statistics

### CODE REVIEW

File: source/docs/user_manual/processing_algs/qgis/vectoranalysis.rst [hunk 1]
```diff
@@ -82,11 +82,11 @@ Outputs
      - 
    * - **Minimum value**
      - ``MIN``
-     - [number]
+     - [same as input]
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vectoranalysis.rst]: @havatv I do not remember (and can't find the discussion) what was decided about using this for field type.

File: source/docs/user_manual/processing_algs/qgis/vectoranalysis.rst [hunk 2]
```diff
@@ -1198,6 +1170,100 @@ Outputs
      - [table]
      - Table containing the statistics
 
+|
+
+Depending on the type of the field being analyzed,
+following statistics are returned for each grouped value:
```
Comment (havatv) [inline source/docs/user_manual/processing_algs/qgis/vectoranalysis.rst]:
```suggestion
the following statistics are returned for each grouped value:
```

File: source/docs/user_manual/processing_algs/qgis/vectoranalysis.rst (hunk 1 above)
Comment (havatv) [reply source/docs/user_manual/processing_algs/qgis/vectoranalysis.rst]: We have only used for vector output until now. I don't think we have made a decision as to using it in other contexts, but I have no objections to using it here.

Comment (DelazJ) [reply source/docs/user_manual/processing_algs/qgis/vectoranalysis.rst]: So be it!

**Pull request merged**